arkouda/pdarrayclass.py

```diff
@@ -1560,3 +1690,156 @@ def argmaxk(pda : pdarray, k : int) -> pdarray:
 
     repMsg = generic_msg("maxk {} {} {}".format(pda.name, k, True))
     return create_pdarray(repMsg)
+
+@typechecked
+def register_pdarray(pda : pdarray, user_defined_name : str) -> pdarray:
+    """
+    Return a pdarray with a user defined name in the arkouda server 
+    so it can be attached to later using attach_pdarray()
+    
+    Parameters
+    ----------
+    pda : str or pdarray
+        the array to register
+    user_defined_name : str
+        user defined name array is to be registered under
+
+    Returns
+    -------
+    pdarray
+        pdarray which points to original input pdarray but is also 
+        registered with user defined name in the arkouda server
+
+
+    Raises
+    ------
+    TypeError
+        Raised if pda is neither a pdarray nor a str or if 
+        user_defined_name is not a str
+
+    See also
+    --------
+    attach_pdarray, unregister_pdarray
+
+    Notes
+    -----
+    Registered names/pdarrays in the server are immune to deletion 
+    until they are unregistered.
+
+    Examples
+    --------
+    >>> a = zeros(100)
+    >>> r_pda = ak.register_pda(a, "my_zeros")
+    >>> # potentially disconnect from server and reconnect to server
+    >>> b = ak.attach_pda("my_zeros")
+    >>> # ...other work...
+    >>> ak.unregister_pda(b)
+    """
+    if not isinstance(user_defined_name, str):
+        raise TypeError("user_defined_name must be a str")
+
+    ret = None
+    if isinstance(pda, pdarray):
+        repMsg = generic_msg("register {} {}".\
+                             format(pda.name, user_defined_name))
+        ret = create_pdarray(repMsg)
+    elif isinstance(pda, str):
+        repMsg = generic_msg("register {} {}".\
+                             format(pda, user_defined_name))        
+        ret = create_pdarray(repMsg)
+    else:
```

Comment:
This raise TypeError clause no longer needed with typechecked annotation.

Reply:
i was hoping you could help me with the type annotations in the function arguments. I think it should be something like
```python
@typechecked
 def register_pdarray(pda : Union[pdarray,str], user_defined_name : str) -> pdarray:
```